meos/src/cbuffer/cbuffer.c

```diff
@@ -127,16 +127,21 @@ cbuffersegm_locate(const Cbuffer *start, const Cbuffer *end,
     if (result2 < 0.0)
       return -1.0;
   }
-  long double result;
+  else
+  {
+    /* If the radii are equal return result1 where
+     * - result1 = -1.0 if gs1 == gs2 == gs, or
+     * - result1 in [0,1] if gs1 != gs2 */
+    return result1;
+  }
```

Comment:
We still need to check that `start->radius == value->radius`, otherwise we should always return `-1`. And what should we return if we have `gs1 == gs2 == gs` and `start->radius == end->radius == value->radius`? This only happens for a constant segment (`start == end`) where `value` is the same as the constant. What do we return in this case for the other types?